pool/src/pool.rs

```diff
@@ -70,11 +71,60 @@ where
 
 	/// Take the first num_to_fetch txs based on insertion order.
 	pub fn prepare_mineable_transactions(&self, num_to_fetch: u32) -> Vec<Transaction> {
-		self.entries
-			.iter()
-			.take(num_to_fetch as usize)
-			.map(|x| x.tx.clone())
-			.collect()
+		// first, group dependent transactions in buckets (vectors), each bucket
+		// is therefore independent from the others
+		// this relies on the entries Vec having parent transactions first
```

Comment:
I assume we can safely rely on this as the pool is sorted internally by tx insertion order.